app/lib/services.rb

```diff
@@ -13,9 +13,14 @@ def self.content_store
   end
 
   def self.cached_content_item(base_path)
-    Rails.cache.fetch("finder-frontend_content_items#{base_path}", expires_in: 5.minutes) do
+    content_item = content_store.content_item(base_path)
+    max_age = content_item.cache_control["max-age"]
+
+    Rails.cache.fetch("finder-frontend_content_items#{base_path}", expires_in: max_age) do
       GovukStatsd.time("content_store.fetch_request_time") do
-        content_store.content_item(base_path).to_h
+        content_item_hash = content_item.to_h
+        content_item_hash["cache_control"] = { "max_age" => max_age, "public" => content_item.cache_control.include?("public") }
```

Comment:
Here could you do `"public" => !content_item.cache_control.private?` to use the existing method.

Mutating the content item hash is a _bit_ of a smell but it's not so bad.

Reply:
ok, I'm trying to use the existing `private` method

Is there another way to add cache_control without mutating the hash?

Reply:
I think mutating the hash is the pragmatic choice, no need to change it 👍